### PR TITLE
Homework 7 MR

### PR DESCRIPTION
**Pull request by Sapphireexe**
Finished and ready to merge:
Task 7.1,
Tasks 7.2 - all tasks,
Task 7.3 - just first subtask. Second subtask with SET method and file is in progress.

### CODE REVIEW

File: HW_7/hw_7_2.js [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+function letterCounter (string) {
+  const arr = [...string.toLowerCase()];
+
+  const numberOfSimilarLetters = arr.map((el, i, arr) => {
+    let counter = 0;
+
+    arr.forEach(el2 => {
+      if (/[a-z]/.test(el2) && el2 === el) {
+        counter++;
+      }
```
Comment (anatoly-karpovich) [inline HW_7/hw_7_2.js:18]: Can be done with filter, but this is ok too

File: HW_7/hw_7_3.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+const fs = require('node:fs');
+let rawData;
+
+try {
+  rawData = fs.readFileSync('./cards.json');
+} catch (err) {
+  console.error('Error reading file cards.json');
+}
+
+const userCardsSet = new Set(JSON.parse(rawData));
```
Comment (anatoly-karpovich) [inline HW_7/hw_7_3.js:43]: Nope, add to set is JSON, i mean string

**Review by anatoly-karpovich — APPROVED** [review]

File: HW_7/hw_7_2.js (hunk 1 above)
Comment (Sapphireexe) [reply HW_7/hw_7_2.js:18]: Will try with filter later, thank you!

File: HW_7/hw_7_3.js (hunk 2 above)
Comment (Sapphireexe) [reply HW_7/hw_7_3.js:43]: Sure, will do this task later. :)

**Pull request merged**